test/lib/jdk/test/lib/util/FileUtils.java

```diff
@@ -377,6 +377,35 @@ public static long getProcessHandleCount() {
         }
     }
 
+    /**
+     * Patches a part of a file.
+     * @param path of file
+     * @param fromLine the first line to patch. This is the number you see in an editor, 1-based.
+     * @param toLine the last line to patch. This is the number you see in an editor, inclusive.
+     * @param from the lines to remove, used to ensure the correct lines are removed. Ignored if null.
+     * @param to the newly added line, can be multiple lines or empty. Cannot be null.
+     * @throws IOException
+     */
+    public static void patch(Path path, int fromLine, int toLine, String from, String to) throws IOException {
```

Comment:
Should this method check whether the `fromLine` and `toLine` are valid values? Things like, negative value or 0 or `toLine` being less than `fromLine`. I'm not familiar with the expectations of test library code - maybe those checks aren't necessary since this is a test util?

Reply:
`lines.remove()` and `lines.subList()` will throw the correct exception. Since you asked, we can add it.

Reply:
Now that we call `subList` at the beginning, I think there's no need to explicitly perform a check.

Reply:
Hello Weijun, the change to this part of the code in context of index checks, looks fine to me now. Thank you for considering it.